README.md

```diff
@@ -37,84 +37,444 @@ lamin {
+    Jun-17 16:30:45.490 [PublishDir-1] DEBUG nextflow.lamin.LaminObserver - Detected previous output artifact jAxjm3Kj39aiPUW30000 (https://staging.laminhub.com/laminlabs/lamindata/artifact/jAxjm3Kj39aiPUW30000)
+    Jun-17 16:30:46.281 [PublishDir-2] DEBUG nextflow.lamin.instance.Instance - Response from createArtifact: [statusCode:200.0, body:[message:Artifact created successfully, artifact:[version:null, is_latest:true, branch:1.0, space:1.0, _aux:null, created_at:2025-06-17T14:30:46.176Z, updated_at:2025-06-17T14:30:46.176Z, uid:j5gTGN8mGWKGwJz40000, key:scratch/temp-nf-lamin/run_20250617_163032/output.gz, description:Output artifact for run 603, storage:32.0, suffix:.gz, kind:null, otype:null, size:86.0, hash:Z8TJDV3943ECoVDiee8OJQ, n_files:null, n_observations:null, _hash_type:md5, run:603.0, schema:null, _key_is_virtual:false, created_by:28.0, _overwrite_versions:false, input_of_runs:[], _subsequent_runs:[], _actions:[]]]]
+    Jun-17 16:30:46.282 [PublishDir-2] DEBUG nextflow.lamin.LaminObserver - Created output artifact j5gTGN8mGWKGwJz40000 (https://staging.laminhub.com/laminlabs/lamindata/artifact/j5gTGN8mGWKGwJz40000)
     DEBUG nextflow.lamin.LaminObserver - onFlowComplete triggered!
-    DEBUG nextflow.lamin.LaminObserver - Finalise Run object:
-      run = ln.Run.get("abcdef123456")
-      run.finished_at = "2025-05-01T15:51:03.312545607+02:00"
+    DEBUG nextflow.lamin.instance.Instance - PATCH updateRecord: core.run, uid=UkLH9Z5eAtcY6iL1bDBG, data=[finished_at:2025-06-17T16:30:46.282525416+02:00, _status_code:0]
+    DEBUG nextflow.lamin.instance.Instance - Response from updateRecord: [id:603.0, uid:UkLH9Z5eAtcY6iL1bDBG, _aux:null, name:loving_darwin, space_id:1.0, _branch_code:1.0, report_id:null, _logfile_id:null, reference:null, transform_id:330.0, created_at:2025-06-17T14:30:37.538033+00:00, created_by_id:28.0, started_at:2025-06-17T14:30:34.908140+00:00, environment_id:null, finished_at:2025-06-17T14:30:46.282525+00:00, _status_code:0.0, reference_type:null, _is_consecutive:null, initiated_by_run_id:null]
+
+</details>
+
+## Settings
+
+The plugin supports the following settings in the `nextflow.config`
+file:
+
+```groovy
+lamin {
+  // The LaminDB instance to use
+  instance = "laminlabs/lamindata"
+  // The API key for the LaminDB instance
+  api_key = secrets.LAMIN_API_KEY
+  // The project name in LaminDB
+  project = "your-lamin-project"
+  // The environment name in LaminDB
+  env = "prod"
+  // The Supabase API URL for the LaminDB instance (if env is set to null)
+  supabase_api_url = "https://your-supabase-api-url.supabase.co"
```

Comment:
Why is this user-facing? This should be hard-coded in the package like it is:

https://github.com/laminlabs/lamindb-setup/blob/747081a49f74b023118a54bc049cfea45ecc8abf/lamindb_setup/core/_hub_client.py#L28-L29

One can deal with different envs and env variables like we do here:

https://github.com/laminlabs/lamindb-setup/blob/747081a49f74b023118a54bc049cfea45ecc8abf/lamindb_setup/core/_hub_client.py#L47-L48

Reply:
It's implemented in the same way in `nf-lamin` as in `lamindb_setup`. Namely, you can set `lamin.env` or `LAMIN_ENV`, but if these are undefined you can define `SUPABASE_API_URL` and `SUPABASE_ANON_KEY` manually.

I think these variables are needed for testing the plugin on a local build, but perhaps you would like me to remove the variables from the readme?

Reply:
Great! And yes, removing them from the README would be good. We could add them to a `CONTRIBUTING` instead once we have that.